liquid-rust-syntax/src/resolve.rs

```diff
@@ -106,6 +106,7 @@ impl<'tcx> Resolver<'tcx> {
             }
             surface::Arg::StrgRef(loc, ty) => Ok(surface::Arg::StrgRef(loc, self.resolve_ty(ty)?)),
             surface::Arg::Ty(ty) => Ok(surface::Arg::Ty(self.resolve_ty(ty)?)),
+            _ => panic!("unexpected: resolve_arg"),
```

Comment:
I prefer to be explicit about the missing case to catch errors if we add more variants.

Reply:
Good point, have fixed -- btw, note that if we had a refinements, we should be able to prove this arm is unreachable...